From 6f460f41a41c7202b93d6951b72d64030d54ca3b Mon Sep 17 00:00:00 2001
From: "Liu,Dalei" <daleix.liu@intel.com>
Date: Mon, 9 Nov 2015 14:30:41 +0800
Subject: [PATCH 01/28] imx: In BTNS get imx227 resolution without ratio

When we are in the platform BTNS, we need some 1:1 ratio
resolution output. So we need to get a nearest resolution
without ratio condition. Otherwise we may get a wrong
resolution.
(e.g. HAL want 640x640, we get 376x656)

Change-Id: I03139307a45bc2a558ff92814d3acbd02ee3d4a3
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-9279
Original-Tracked-On: #1604036344
Signed-off-by: Liu,Dalei <daleix.liu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/83892
Approver: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Build: Dessert M COHO (SC)
Build: Dessert M BXT RVP (SC)
Integrator: WeiX Meng <weix.meng@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Signed-off-by: Yang Fan F <fan.f.yang@intel.com>
Reviewed-on: https://android.intel.com:443/440487
---
 drivers/external_drivers/camera/drivers/media/i2c/imx/imx.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/imx/imx.c b/drivers/external_drivers/camera/drivers/media/i2c/imx/imx.c
index d22aa02..9ac7276 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/imx/imx.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/imx/imx.c
@@ -1625,7 +1625,7 @@ retry:
 
 	/*
 	 * FIXME!
-	 * only IMX135 for Saltbay use this algorithm
+	 * only IMX135 for Saltbay and IMX227 use this algorithm
 	 */
 	if (idx == -1 && again == true && dev->new_res_sel_method) {
 		again = false;
@@ -2099,6 +2099,7 @@ static int __update_imx_device_settings(struct imx_device *dev, u16 sensor_id)
 		dev->mode_tables = &imx_sets[IMX227_SAND];
 		dev->vcm_driver = NULL;
 		dev->otp_driver = &imx_otps[IMX227_SAND];
+		dev->new_res_sel_method = true;
 		return 0;
 	case IMX132_ID:
 		dev->mode_tables = &imx_sets[IMX132_SALTBAY];
-- 
1.9.1

